database/index.go

```diff
@@ -48,6 +48,53 @@ func term(s string) string {
 
 var httpPat = regexp.MustCompile(`https?://\S+`)
 
+func collectSynopsisTerms(terms map[string]bool, synopsis string) {
+
+	synopsis = httpPat.ReplaceAllLiteralString(synopsis, "")
+
+	fields := strings.FieldsFunc(synopsis, isTermSep)
+	for i := range fields {
+		fields[i] = strings.ToLower(fields[i])
+	}
+
+	if len(fields) > 0 && fields[0] == "the" {
+		fields = fields[1:]
+	}
+
+	// Ignore boilerplate in the following common patterns:
+	//      Package foo ...
+	//      Package foo implements ...
+	//      Package foo provides ...
+	//      Package foo contains ...
+	//      Command foo ...
+	if len(fields) > 0 {
```

Comment:
I wrote a test case for this, but the test case is wrong. Oops. It's worth removing the "the foo package implements".  I'll update the code.